generate_toc.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# adapted from https://github.com/npryce/adr-tools
+# a combination of a few scripts, _adr_title, _adr_generate_toc
+
+set -e
+
+args=$(getopt i:o:p: $*)
+set -- $args
+
+link_prefix=./decisions/
+
+eval "$(adr config)"
+
+cat <<EOF
+# Architecture Decision Records
+EOF
+
+if [ ! -z $intro ]
```

Comment:
This break if `$intro` contains whitespace, and `! -z` is the same as `-n`.
```suggestion
if [ -n "$intro" ];
```